File: 162-find-peak-element/162-find-peak-element.go
func findPeakElement(nums []int) int {
    l := 0
    r := len(nums) - 1
    for l < r {
        mid := ( l + r ) / 2
        if nums[mid] > nums[mid + 1] {
            if mid == 0 || nums[mid] > nums[mid - 1] {
                return mid
            }
            r = mid
        } else {
            l = mid + 1
        }
    }
    return l
}

File: 162-find-peak-element/162-find-peak-element.py
class Solution:
    """
    Use Binary Search to find peak elem
    """
    def findPeakElement(self, nums: List[int]) -> int:
        num_len = len(nums)
        l, r = 0, num_len - 1
        while l < r:
            mid = (l + r) >> 1
            if nums[mid] > nums[mid + 1]: r = mid
            else: l = mid + 1
        return l